src/function/compression_config.cpp

```diff
@@ -54,6 +53,7 @@ static optional_ptr<CompressionFunction> LoadCompressionFunction(CompressionFunc
 			}
 			// the type is supported: create the function and insert it into the set
 			auto function = method.get_function(info.GetPhysicalType());
```

Comment:
Should we remove the call to `supports_type` here entirely and make sure this function is only called in one place?

Otherwise we might end up with a situation where loading a database with block size 16KB will prevent using compression functions with other attached databases that do support it.

Reply:
Maybe we can also write a test for this where we start off attaching a database with block size = 16KB, then attach one with block size = 256KB, and then verify that we can use the compression function that is disabled with block size = 16KB in the database that has the higher block size?

Reply:
> Should we remove the call to supports_type here entirely and make sure this function is only called in one place?

How would we manage the `set` holding all compression functions? `set.functions[type].insert(make_pair(info.GetPhysicalType(), function));` matches them with a physical type, and if we skip the `type_is_supported` check, then we'll have mismatching set insertions?

> Otherwise we might end up with a situation where loading a database with block size 16KB will prevent using compression functions with other attached databases that do support it.

This should not happen, as we won't be able to load a matching function, right? We'll then loop over all `internal_compression_methods` and reinsert that function with the matching physical type.